src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/NFloatTests.cs

```diff
@@ -70,15 +69,38 @@ public void EqualsTest(NFloat f1, object obj, bool expected)
             Assert.Equal(expected, f1.Equals(obj));
         }
 
-        [Theory]
+        [Fact]
+        public void NaNEqualsTest()
+        {
+            NFloat f1 = new NFloat(float.NaN);
+            NFloat f2 = new NFloat(float.NaN);
+            Assert.Equal(f1.Value == f2.Value, f1.Equals(f2));
```

Comment:
This seems incorrect. For `floating-point`, `==` and `Equals` are different due to needing to follow IEEE vs .NET equality rules.

I would think that `NFloat` should also follow that for the limited set of APIs it does expose.